### PR TITLE
[CI/CD]: Review coverage of accessibility checks in GI Bill Comparison Tool end-to-end tests #7373

### PR DESCRIPTION
**Pull request by zurbergram**
## Description

https://github.com/department-of-veterans-affairs/va.gov-team/issues/7373

- added ids to GI accordion components
- updated test data (also moved into json files from `src/applications/gi/tests/gibct-helpers.js` to allow for easier updating in future from relevant staging endpoints)
- for`src/applications/gi/tests/00-required.e2e.spec.js` test added more testing so will fail if goes to error page
- created `src/applications/gi/tests/02-profile-error.e2e.spec.js` test to axe check the error page
- created `src/applications/gi/tests/03-vet-tec.e2e.spec.js` test to test VETTEC flow 

## Testing done
` yarn test:e2e src/applications/gi/tests`

## Acceptance criteria
- [ ] Front-end team member(s) have reviewed end-to-end tests for axe checks
- [ ] Ensure provider view e2e test resolves to an actual provider
- [ ] Expand the `Edit eligibility details` accordion on the provider page, then run the axe check
- [ ] Open an modal window on the provider detail view, and run another axe check
- [ ] If possible, maintain the existing error page for provider, and run that axe check
- [ ] FE team has consulted with accessibility specialist in cases where there are high numbers of modals, accordions, other hidden content that could slow down e2e test runs.
- [ ] No axe `violations` appear in the plugin console. These will break the CI/CD build.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/tests/00-required.e2e.spec.js [hunk 1]
```diff
@@ -1,37 +1,65 @@
...
   client
-    .click('#search-button')
-    .waitForElementVisible('.search-page', Timeouts.normal)
-    .axeCheck('.main');
-  client
-    .waitForElementVisible('.search-result a', Timeouts.normal)
-    .click('.search-result a')
     .waitForElementVisible('.profile-page', Timeouts.normal)
-    .axeCheck('.main');
+    .axeCheck('.main'); // commented out until 7727 is fixed
```
Comment (1Copenut) [inline src/applications/gi/tests/00-required.e2e.spec.js:40]: When you're ready to move this PR out of draft, let's move this `axeCheck` to after the accordions are opened. We're looking to increase test coverage, and axe doesn't check things that are hidden using `display: none` or other CSS means.

File: src/applications/gi/tests/03-vet-tec.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  GiHelpers.selectFirstSearchResult(client);
+
+  // Profile Page
+  GiHelpers.expectLocation(
+    client,
+    `/profile/${vetTecAttributes.facility_code}/`,
+  );
+  client
+    .waitForElementVisible('.profile-page', Timeouts.normal)
+    .axeCheck('.main'); // commented out until 7727 is fixed
```
Comment (1Copenut) [inline src/applications/gi/tests/03-vet-tec.e2e.spec.js:36]: Same comment about moving the axe check after the accordions have been opened.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: src/applications/gi/tests/00-required.e2e.spec.js (hunk 1 above)
Comment (zurbergram) [reply src/applications/gi/tests/00-required.e2e.spec.js:40]: By default all the major section accordions are open when loading a profile page
See https://staging.va.gov/gi-bill-comparison-tool/profile/14900416

The "Eligibility Details" section is expanded in `GiHelpers.editEligibilityDetails(client);` and an axe check is performed on that section specifically after it is expanded.

File: src/applications/gi/tests/03-vet-tec.e2e.spec.js (hunk 2 above)
Comment (zurbergram) [reply src/applications/gi/tests/03-vet-tec.e2e.spec.js:36]: By default all accordions are expanded when loading a VET TEC profile page
see https://staging.va.gov/gi-bill-comparison-tool/profile/2V000105/HACK%20REACTOR%20SOFTWARE%20ENGINEERING%20ONLINE

Comment (1Copenut) [reply src/applications/gi/tests/03-vet-tec.e2e.spec.js:36]: 👍 Cool, thanks for confirming.

**Pull request merged**